jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -223,6 +228,19 @@ def _on_add_data_to_viewer(self, msg):
         if isinstance(msg.data, glue_core_data):
             data_obj = None
 
+        # sync viewer and UC plugin units when new data is loaded. this is necessary, for
+        # example, when all data is unloaded but unit selections are set in the UC plugin,
+        # and new data is loaded which should use those units
+        if len(self.spectral_unit_selected):
+            if hasattr(viewer.state, 'x_display_unit') and len(self.spectral_unit_selected):
+                if viewer.state.x_display_unit != self.spectral_unit_selected:
+                    xunit = _valid_glue_display_unit(self.spectral_unit.selected, viewer, 'x')
+                    viewer.state.x_display_unit = xunit
+                    viewer.set_plot_axes()
+        if len(self.spectral_y_unit) and hasattr(viewer.state, 'y_display_unit'):
+            if viewer.state.y_display_unit != self.spectral_y_unit:
```

Comment:
also could probably be a single if-statement, but again would probably require a line-break